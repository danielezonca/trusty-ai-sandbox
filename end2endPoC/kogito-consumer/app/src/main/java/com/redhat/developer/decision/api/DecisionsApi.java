package com.redhat.developer.decision.api;

import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;

import javax.inject.Inject;
import javax.validation.constraints.NotNull;
import javax.ws.rs.DefaultValue;
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

import com.redhat.developer.database.IEventStorage;
import com.redhat.developer.decision.responses.DecisionDetailResponse;
import com.redhat.developer.decision.responses.DecisionInputsResponse;
import com.redhat.developer.decision.responses.DecisionOutputsResponse;
import com.redhat.developer.decision.responses.ExecutionHeaderResponse;
import com.redhat.developer.decision.responses.ExecutionResponse;
import com.redhat.developer.decision.storage.model.DMNResultModel;
import org.jboss.resteasy.annotations.jaxrs.PathParam;
import org.jboss.resteasy.annotations.jaxrs.QueryParam;

@Path("/executions")
public class DecisionsApi {

    @Inject
    IEventStorage storageService;

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public ExecutionResponse getExecutions(@DefaultValue("yesterday") @QueryParam("from") String from,
                                           @DefaultValue("now") @QueryParam("to") String to,
                                           @DefaultValue("100") @QueryParam("limit") int limit,
                                           @DefaultValue("0") @QueryParam("offset") int offset) {
        // Testing hack
        if (from.equals("yesterday")){
            from = java.time.LocalDateTime.now().minusDays(1).toString();
        }

        List<DMNResultModel> results = storageService.getDecisions(from, to);
        int totalResults = results.size();
        if (totalResults >= limit){
            results.sort(Comparator.comparing(DMNResultModel::getExecutionDate));
            results = results.subList(offset, Math.min(results.size(), offset + limit));
        }

        List<ExecutionHeaderResponse> executionResponses = new ArrayList<>();
        results.forEach(x -> executionResponses.add(buildHeaderResponse(x)));
        return new ExecutionResponse(totalResults, limit, offset, executionResponses);
    }

    @GET
    @Path("/{key}")
    @Produces(MediaType.APPLICATION_JSON)
    public DecisionDetailResponse getExecutionByKey(@PathParam("key") String key) {
        DMNResultModel result = storageService.getEvent(key).data.result;
        return new DecisionDetailResponse(buildHeaderResponse(result));
    }

    @GET
    @Path("/{key}/inputs")
    @Produces(MediaType.APPLICATION_JSON)
    public DecisionInputsResponse getExecutionInputs(@PathParam("key") String key) {
        DMNResultModel result = storageService.getEvent(key).data.result;
        return new DecisionInputsResponse(buildHeaderResponse(result), result.context);
    }

    @GET
    @Path("/{key}/outcomes")
    @Produces(MediaType.APPLICATION_JSON)
    public DecisionOutputsResponse getExecutionOutcome(@PathParam("key") String key) {
        DMNResultModel result = storageService.getEvent(key).data.result;
        return new DecisionOutputsResponse(buildHeaderResponse(result), result.decisions);
    }

    private ExecutionHeaderResponse buildHeaderResponse(DMNResultModel result) {
        return new ExecutionHeaderResponse(result.executionId, result.executionDate, result.decisions.stream().anyMatch(y -> y.hasErrors == true), "testUser");
    }
}
