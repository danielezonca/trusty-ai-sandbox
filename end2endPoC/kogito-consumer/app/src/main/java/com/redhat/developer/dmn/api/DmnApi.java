package com.redhat.developer.dmn.api;

import java.util.List;
import java.util.stream.Collectors;

import javax.enterprise.context.RequestScoped;
import javax.inject.Inject;
import javax.ws.rs.Consumes;
import javax.ws.rs.GET;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

import com.redhat.developer.database.IStorageManager;
import com.redhat.developer.dmn.IDmnService;
import com.redhat.developer.dmn.requests.EvaluationRequestBody;
import com.redhat.developer.dmn.requests.NewDmnModelRequest;
import com.redhat.developer.dmn.responses.AvailableModelsResponse;
import com.redhat.developer.dmn.responses.EvaluationResponse;
import com.redhat.developer.dmn.responses.FullModelResponse;
import com.redhat.developer.dmn.responses.ModelDetail;
import com.redhat.developer.dmn.responses.inputs.ModelInputResponse;
import com.redhat.developer.dmn.storage.dto.DmnModel;
import com.redhat.developer.dmn.utils.MyMD5;
import org.eclipse.microprofile.openapi.annotations.Operation;
import org.eclipse.microprofile.openapi.annotations.enums.SchemaType;
import org.eclipse.microprofile.openapi.annotations.media.Content;
import org.eclipse.microprofile.openapi.annotations.media.Schema;
import org.eclipse.microprofile.openapi.annotations.parameters.Parameter;
import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
import org.eclipse.microprofile.openapi.annotations.responses.APIResponses;
import org.jboss.resteasy.annotations.jaxrs.PathParam;

@Path("/models")
@RequestScoped
public class DmnApi {

    @Inject
    IStorageManager storageManager;

    @Inject
    IDmnService dmnService;

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public AvailableModelsResponse getModels() {
        List<DmnModel> ids = storageManager.getModelIds();
        return new AvailableModelsResponse(ids.stream().map(x -> ModelDetail.fromStorageModel(x)).collect(Collectors.toList()));
    }

    @POST
    @Produces(MediaType.TEXT_PLAIN)
    @Consumes(MediaType.APPLICATION_JSON)
    public Response setModel(NewDmnModelRequest request) {
        DmnModel model = DmnModel.fromNewDmnModelRequest(request, MyMD5.getMd5(request.nameSpace + request.name));
        storageManager.storeModel(model);
        return Response.ok(model.modelId).build();
    }


    @GET
    @Path("/{id}")
    @Produces(MediaType.APPLICATION_JSON)
    public FullModelResponse getModelById(@PathParam("id") String id) {
        DmnModel model = storageManager.getModel(id);
        return FullModelResponse.fromStorageModel(model);
    }

    @GET
    @Path("/{id}/inputs")
    @APIResponses(value = {
            @APIResponse(description = "Returns the input structure of the dmn model.", responseCode = "200", content = @Content(mediaType = MediaType.APPLICATION_JSON, schema = @Schema(type = SchemaType.OBJECT, implementation = ModelInputResponse.class))),
            @APIResponse(description = "Bad Request", responseCode = "400", content = @Content(mediaType = MediaType.TEXT_PLAIN))
    }
    )
    @Operation(summary = "Gets the structure of the dmn inputs.", description = "Gets the structure of the dmn inputs.")
    @Produces(MediaType.APPLICATION_JSON)
    public Response getModelInputs(@PathParam("id") String id) {
        return Response.ok(new ModelInputResponse()).build();
    }

    @POST
    @Path("/{id}/evaluate")
    @Produces(MediaType.APPLICATION_JSON)
    @Consumes(MediaType.APPLICATION_JSON)
    public EvaluationResponse evaluateDecision(@Parameter(
            name = "id",
            description = "ID of the DMN Model to evaluate.",
            required = true,
            schema = @Schema(implementation = String.class)
    ) @PathParam("id") String id, EvaluationRequestBody inputs) {
        Object o = dmnService.evaluateModel(id, inputs.inputs);
        return new EvaluationResponse(o);
    }
}