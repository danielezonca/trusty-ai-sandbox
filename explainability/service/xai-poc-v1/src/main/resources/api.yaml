swagger: "2.0"
info:
  description: "Trusty XAI service description."
  version: "1.0.0"
  title: "TrustyXAI"
  termsOfService: "http://kie.org/trusty/"
  contact:
    email: "tteofili@redhat.com"
  license:
    name: "Apache 2.0"
    url: "http://www.apache.org/licenses/LICENSE-2.0.html"
host: "kie.org"
basePath: "/xai"
tags:
- name: "local"
  description: "Local explanations"
  externalDocs:
    description: "Find out more"
    url: "http://kie.org"
- name: "global"
  description: "Global explanations"
  externalDocs:
    description: "Find out more"
    url: "http://kie.org"
schemes:
- "https"
- "http"
paths:
  /local/saliency/lime:
    post:
      tags:
      - "local"
      summary: "Execute LIME explanation method on a prediction"
      description: ""
      operationId: "lime"
      consumes:
      - "application/json"
      produces:
      - "application/json"
      parameters:
      - in: "body"
        name: "body"
        description: "Prediction that needs to be explained"
        required: true
        schema:
          $ref: "m2x.yaml#/definitions/Prediction"
      responses:
        200:
          description: "successful explanation"
          schema:
            $ref: "#/definitions/Saliency"
      security:
      - trusty_auth:
        - "write:exp"
        - "read:exp"
  /global/tabular/pdp:
    post:
      tags:
      - "global"
      summary: "Generate PDP explanation method on a model"
      description: ""
      operationId: "pdp"
      consumes:
      - "application/json"
      produces:
      - "application/json"
      parameters:
      - in: "body"
        name: "body"
        description: "Model that needs to be explained"
        required: true
        schema:
          $ref: "m2x.yaml#/definitions/ModelInfo"
      responses:
        200:
          description: "successful explanation"
          schema:
            $ref: "#/definitions/TabularData"
      security:
      - trusty_auth:
        - "write:exp"
        - "read:exp"
securityDefinitions:
  trusty_auth:
    type: "oauth2"
    authorizationUrl: "http://kie.org/oauth/dialog"
    flow: "implicit"
    scopes:
      write:exp: "execute explanations"
      read:exp: "read explanations"
  api_key:
    type: "apiKey"
    name: "api_key"
    in: "header"
definitions:
  TabularData:
    type: "object"
    properties:
      x-axis:
        type: "array"
        items:
          type: "number"
      y-axis:
        type: "array"
        items:
          type: "number"
  Saliency:
   type: "object"
   properties:
    featureImportances:
      type: "array"
      items:
        $ref: "#/definitions/FeatureImportance"
  FeatureImportance:
    type: "object"
    properties:
      feature:
        $ref: "m2x.yaml#/definitions/Feature"
      score:
        type : "number"