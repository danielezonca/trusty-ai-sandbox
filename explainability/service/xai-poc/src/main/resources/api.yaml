swagger: "2.0"
info:
  description: "Trusty XAI service description."
  version: "1.0.0"
  title: "TrustyXAI"
  termsOfService: "http://kie.org/trusty/"
  contact:
    email: "tteofili@redhat.com"
  license:
    name: "Apache 2.0"
    url: "http://www.apache.org/licenses/LICENSE-2.0.html"
host: "kie.org"
basePath: "/xai"
tags:
- name: "local"
  description: "Local explanations"
  externalDocs:
    description: "Find out more"
    url: "http://kie.org"
- name: "global"
  description: "Global explanations"
  externalDocs:
    description: "Find out more"
    url: "http://kie.org"
schemes:
- "https"
- "http"
paths:
  /local/saliency/lime:
    post:
      tags:
      - "local"
      summary: "Execute LIME explanation method on a prediction"
      description: ""
      operationId: "lime"
      consumes:
      - "application/json"
      - "application/xml"
      produces:
      - "application/xml"
      - "application/json"
      parameters:
      - in: "body"
        name: "body"
        description: "Prediction that needs to be explained"
        required: true
        schema:
          $ref: "#/definitions/Prediction"
      responses:
        200:
          description: "successful explanation"
          schema:
            $ref: "#/definitions/Saliency"
      security:
      - trusty_auth:
        - "write:exp"
        - "read:exp"
  /global/tabular/pdp:
    post:
      tags:
      - "global"
      summary: "Generate PDP explanation method on a model"
      description: ""
      operationId: "pdp"
      consumes:
      - "application/json"
      produces:
      - "application/json"
      parameters:
      - in: "body"
        name: "body"
        description: "Model that needs to be explained"
        required: true
        schema:
          $ref: "#/definitions/ModelInfo"
      responses:
        200:
          description: "successful explanation"
          schema:
            $ref: "#/definitions/TabularData"
      security:
      - trusty_auth:
        - "write:exp"
        - "read:exp"
securityDefinitions:
  trusty_auth:
    type: "oauth2"
    authorizationUrl: "http://kie.org/oauth/dialog"
    flow: "implicit"
    scopes:
      write:exp: "execute explanations"
      read:exp: "read explanations"
  api_key:
    type: "apiKey"
    name: "api_key"
    in: "header"
definitions:
  TabularData:
    type: "object"
    properties:
      x-axis:
        type: "array"
        items:
          type: "number"
      y-axis:
        type: "array"
        items:
          type: "number"
  Saliency:
   type: "object"
   properties:
    featureImportances:
      type: "array"
      items:
        $ref: "#/definitions/FeatureImportance"
  FeatureImportance:
    type: "object"
    properties:
      feature:
        $ref: "#/definitions/Feature"
      score:
        type : "number"
  Feature:
    type: "object"
    properties:
      name:
        type: "string"
      type:
        type: "string"
        enum:
          - "quantitative"
          - "categorical"
          - "ordinal"
      value:
        type: "string"
  PredictionInput:
    type: "object"
    properties:
      features:
        type: "array"
        items:
          $ref: "#/definitions/Feature"
  PredictionOutput:
    type: "object"
    properties:
      outputs:
        type: "array"
        items:
          $ref: "#/definitions/Output"
  Output:
    type: "object"
    properties:
      label:
        type: "string"
      score:
        type: "number"
  FeatureDistribution:
    type: "object"
    properties:
      min:
        type: "number"
      max:
        type: "number"
      mean:
        type: "number"
      stdDev:
        type: "number"
  DataDistribution:
    type: "object"
    properties:
      featureDistributions:
        type: "array"
        items:
          $ref: "#/definitions/FeatureDistribution"
  ModelInfo:
    type: "object"
    properties:
      uuid:
        type: "string"
      inputShape:
        type: "integer"
      outputShape:
        type: "integer"
      name:
        type: "string"
      version:
        type: "string"
      taskType:
        type: "string"
        enum:
          - "classification"
          - "regression"
      dataDistribution:
        $ref: "#/definitions/DataDistribution"
  Prediction:
    type: "object"
    properties:
      predictionInput:
        $ref: "#/definitions/PredictionInput"
      predictionOutput:
        $ref: "#/definitions/PredictionOutput"
      modelInfo:
        $ref: "#/definitions/ModelInfo"
externalDocs:
  description: "Find out more about Swagger"
  url: "http://swagger.io"