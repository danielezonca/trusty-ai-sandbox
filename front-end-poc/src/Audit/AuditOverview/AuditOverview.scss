.execution-status-badge {
  margin-right: 8px;
  &--success {
    color: var(--pf-global--success-color--100);
  }
  &--failure {
    color: var(--pf-global--danger-color--100);
  }
}