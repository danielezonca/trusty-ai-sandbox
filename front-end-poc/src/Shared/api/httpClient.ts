import axios, { AxiosRequestConfig, CancelTokenSource } from "axios";
import { toast, Slide } from "react-toastify";
import ErrorNotification from "../components/ErrorNotification/ErrorNotification";

const httpClient = axios.create({
  baseURL: "http://localhost:1336",
  timeout: 5000,
  headers: {},
});

httpClient.interceptors.response.use(
  (response) => response,
  (error) => {
    if (error.response) {
      const { status, statusText, data } = error.response;
      if (status === 500) {
        toast(({ closeToast }) => ErrorNotification({ closeToast, message: statusText, details: data.details }), {
          transition: Slide,
          autoClose: 5000,
          hideProgressBar: true,
        });
      }
    } else {
      return Promise.reject(error);
    }
  }
);

/**
 * first implementation of a repeated call canceler
 * basically it manipulates the conf object adding canceler
 * and calling it when repeated calls happens.
 * at some point it will need to check for an identifier
 * if parallel requests need canceling
 * */
let call: CancelTokenSource;
const callOnce = (config: AxiosRequestConfig) => {
  if (call) {
    call.cancel("Request superseded");
  }
  call = axios.CancelToken.source();

  config.cancelToken = call.token;
  return httpClient(config);
};

const isCancelledRequest = axios.isCancel;

export { httpClient, callOnce, isCancelledRequest };
