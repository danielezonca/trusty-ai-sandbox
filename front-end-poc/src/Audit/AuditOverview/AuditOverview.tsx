import React, { useEffect, useState } from "react";
import {
  List,
  ListItem,
  ListVariant,
  PageSection,
  PageSectionVariants,
  Text,
  TextContent,
  Title,
  Toolbar,
  ToolbarContent,
  ToolbarItem,
} from "@patternfly/react-core";
import { Link } from "react-router-dom";
import { getExecutions } from "../../Shared/api/audit.api";
import { IExecution } from "../types";
import "./AuditOverview.scss";
import PaginationContainer from "../PaginationContainer/PaginationContainer";
import SkeletonStripe from "../../Shared/skeletons/SkeletonStripe/SkeletonStripe";
import AuditToolbar from "../AuditToolbar/AuditToolbar";
import ExecutionTable from "../ExecutionTable/ExecutionTable";
import { RemoteData } from "../../Shared/types";

const AuditOverview = () => {
  const oneMonthAgo = new Date();
  oneMonthAgo.setMonth(oneMonthAgo.getMonth() - 1);
  const [data, setData] = useState<RemoteData<Error, IExecution[]>>({
    status: "NOT_ASKED",
  });
  const [searchString, setSearchString] = useState("");
  const [latestSearches, setLatestSearches] = useState<string[] | null>(null);
  const [fromDate, setFromDate] = useState(oneMonthAgo.toISOString().substr(0, 10));
  const [toDate, setToDate] = useState(new Date().toISOString().substr(0, 10));
  const [page, setPage] = useState(1);
  const [pageSize, setPageSize] = useState(10);
  const [total, setTotal] = useState(0);

  useEffect(() => {
    let didMount = true;
    setData({ status: "LOADING" });
    getExecutions(searchString, fromDate, toDate, pageSize, pageSize * (page - 1))
      .then((response) => {
        if (didMount) {
          setData({
            status: "SUCCESS",
            data: response.data.headers,
          });
          setTotal(response.data.total);
          // temporary solution: for demo purposes we display the first 3 executions here
          if (latestSearches === null) {
            let searches = [];
            let maxSearches = Math.min(3, response.data.total);
            for (let i = 0; i < maxSearches; i++) {
              searches.push(response.data.headers[i].executionId);
            }
            setLatestSearches(searches);
          }
        }
      })
      .catch((error) => {
        setData({ status: "FAILURE", error: error });
      });
    return () => {
      didMount = false;
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [searchString, fromDate, toDate, page, pageSize]);

  return (
    <>
      <PageSection variant={PageSectionVariants.light}>
        <TextContent>
          <Title size="3xl" headingLevel="h2">
            Audit Investigation
          </Title>
          <Text component="p">Here you can retrieve all the available information about past cases</Text>
        </TextContent>
      </PageSection>
      <PageSection style={{ minHeight: "50em" }} isFilled={true}>
        <div style={{ marginBottom: "var(--pf-global--spacer--lg)" }}>
          <List variant={ListVariant.inline}>
            <ListItem>Last Opened:</ListItem>
            {latestSearches === null && <SkeletonStripe isInline={true} customStyle={{ height: "inherit" }} />}
            {latestSearches && latestSearches.length === 0 && (
              <span>
                <em>None</em>
              </span>
            )}
            {latestSearches &&
              latestSearches.length > 0 &&
              latestSearches.map((item, index) => {
                let latestSearchId;
                if (item.toString().indexOf("-") > -1) {
                  let splitted = item.split("-");
                  latestSearchId = splitted[splitted.length - 1];
                } else latestSearchId = item;
                return (
                  <ListItem key={`row-${index}`}>
                    <Link to={`/audit/decision/${item}`}>#{latestSearchId}</Link>
                  </ListItem>
                );
              })}
          </List>
        </div>
        <AuditToolbar
          setSearchString={setSearchString}
          fromDate={fromDate}
          setFromDate={setFromDate}
          toDate={toDate}
          setToDate={setToDate}
          total={total}
          pageSize={pageSize}
          page={page}
          setPage={setPage}
          setPageSize={setPageSize}
        />

        <ExecutionTable data={data} />

        <Toolbar id="audit-list-bottom-toolbar">
          <ToolbarContent>
            <ToolbarItem variant="pagination">
              <PaginationContainer
                total={total}
                page={page}
                pageSize={pageSize}
                onSetPage={setPage}
                onSetPageSize={setPageSize}
                paginationId="audit-list-bottom-pagination"
              />
            </ToolbarItem>
          </ToolbarContent>
        </Toolbar>
      </PageSection>
    </>
  );
};
export default AuditOverview;
