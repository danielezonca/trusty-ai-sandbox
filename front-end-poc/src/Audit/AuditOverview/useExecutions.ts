import { useCallback, useEffect, useState } from "react";
import { getExecutions } from "../../Shared/api/audit.api";
import { RemoteData } from "../../Shared/types";
import { IExecutions } from "../types";
import { isCancelledRequest } from "../../Shared/api/httpClient";

const useExecutions = (searchString: string, from: string, to: string, limit: number, offset: number) => {
  const [executions, setExecutions] = useState<RemoteData<Error, IExecutions>>({
    status: "NOT_ASKED",
  });

  const loadExecutions = useCallback(() => {
    let isMounted = true;
    setExecutions({ status: "LOADING" });
    getExecutions(searchString, from, to, limit, offset)
      .then((response) => {
        if (isMounted) {
          setExecutions({ status: "SUCCESS", data: response.data });
        }
      })
      .catch((error) => {
        if (!isCancelledRequest(error)) {
          setExecutions({ status: "FAILURE", error: error });
        }
      });
    return () => {
      isMounted = false;
    };
  }, [searchString, from, to, limit, offset]);

  useEffect(() => {
    loadExecutions();
  }, [searchString, from, to, limit, offset, loadExecutions]);

  return { loadExecutions, executions };
};

export default useExecutions;
