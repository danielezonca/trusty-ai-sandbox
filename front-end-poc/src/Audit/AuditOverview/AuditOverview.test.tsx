import React from "react";
import { mount, shallow } from "enzyme";
import AuditOverview from "./AuditOverview";
import * as api from "../../Shared/api/audit.api";
import { BrowserRouter } from "react-router-dom";
import { act } from "react-dom/test-utils";

const executions = {
  data: {
    total: 28,
    limit: 50,
    offset: 0,
    headers: [
      {
        executionId: "b2b0ed8d-c1e2-46b5-ad4f-3ac54ff4beae",
        executionDate: "2020-06-01T12:33:57+0000",
        executionSucceeded: true,
        executorName: "testUser",
        executedModelName: "LoanEligibility",
        executionType: "DECISION",
      },
      {
        executionId: "023a0d79-2be6-4ec8-9ef7-99a6796cb319",
        executionDate: "2020-06-01T12:33:57+0000",
        executionSucceeded: true,
        executorName: "testUser",
        executedModelName: "LoanEligibility",
        executionType: "DECISION",
      },
      {
        executionId: "3a5d4a4e-7c5a-4ce7-85de-6024fbf1da39",
        executionDate: "2020-06-01T12:33:56+0000",
        executionSucceeded: true,
        executorName: "testUser",
        executedModelName: "LoanEligibility",
        executionType: "DECISION",
      },
      {
        executionId: "a4e0b8e8-9a6d-4a8e-ad5a-54e5c654a248",
        executionDate: "2020-06-01T12:33:23+0000",
        executionSucceeded: true,
        executorName: "testUser",
        executedModelName: "fraud-scoring",
        executionType: "DECISION",
      },
      {
        executionId: "f08adc80-2c2d-43f4-801c-4f08e10820a0",
        executionDate: "2020-06-01T12:33:18+0000",
        executionSucceeded: true,
        executorName: "testUser",
        executedModelName: "fraud-scoring",
        executionType: "DECISION",
      },
    ],
  },
};
const flushPromises = () => new Promise(setImmediate);
const apiMock = jest.spyOn(api, "getExecutions");
// @ts-ignore
apiMock.mockImplementation(() => Promise.resolve(executions));

describe("Audit overview", () => {
  test("renders", () => {
    const wrapper = shallow(<AuditOverview />);
    expect(wrapper).toMatchSnapshot();
  });

  test("loads a list of executions from the last month", async () => {
    const wrapper = mount(
      <BrowserRouter>
        <AuditOverview />
      </BrowserRouter>
    );
    const oneMonthAgo = new Date();
    oneMonthAgo.setMonth(oneMonthAgo.getMonth() - 1);
    const fromDate = oneMonthAgo.toISOString().substr(0, 10);
    const toDate = new Date().toISOString().substr(0, 10);

    await act(async () => {
      await flushPromises();
      wrapper.update();
    });

    expect(apiMock).toHaveBeenCalledTimes(1);
    expect(apiMock).toHaveBeenCalledWith("", fromDate, toDate, 10, 0);
    expect(wrapper.find("ExecutionTable").props().data).toStrictEqual({
      status: "SUCCESS",
      data: executions.data.headers,
    });
  });
});
