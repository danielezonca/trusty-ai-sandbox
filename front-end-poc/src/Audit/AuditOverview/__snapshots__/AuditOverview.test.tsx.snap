// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audit overview renders 1`] = `
<Fragment>
  <PageSection
    variant="light"
  >
    <TextContent>
      <Title
        headingLevel="h1"
        size="4xl"
      >
        Audit Investigation
      </Title>
      <Text
        component="p"
      >
        Here you can retrieve all the available information about past cases
      </Text>
    </TextContent>
  </PageSection>
  <PageSection
    isFilled={true}
    style={
      Object {
        "minHeight": "50em",
      }
    }
  >
    <div
      style={
        Object {
          "marginBottom": "var(--pf-global--spacer--lg)",
        }
      }
    >
      <List
        variant="inline"
      >
        <ListItem>
          Last Opened:
        </ListItem>
        <SkeletonInlineStripe
          customStyle={
            Object {
              "height": "inherit",
            }
          }
        />
      </List>
    </div>
    <AuditToolbar
      fromDate="2020-05-05"
      page={1}
      pageSize={10}
      setFromDate={[Function]}
      setPage={[Function]}
      setPageSize={[Function]}
      setSearchString={[Function]}
      setToDate={[Function]}
      toDate="2020-06-05"
      total={0}
    />
    <ExecutionTable
      data={
        Object {
          "status": "NOT_ASKED",
        }
      }
    />
    <DataToolbar
      id="audit-list-bottom-toolbar"
      style={
        Object {
          "marginTop": "var(--pf-global--spacer--lg)",
        }
      }
    >
      <DataToolbarContent
        breakpointMods={Array []}
        isExpanded={false}
        showClearFiltersButton={false}
      >
        <DataToolbarItem
          variant="pagination"
        >
          <PaginationContainer
            onSetPage={[Function]}
            onSetPageSize={[Function]}
            page={1}
            pageSize={10}
            paginationId="audit-list-bottom-pagination"
            total={0}
          />
        </DataToolbarItem>
      </DataToolbarContent>
    </DataToolbar>
  </PageSection>
</Fragment>
`;
